portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -48,29 +50,38 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String comment = request.getParameter("Comment:");
-    long timeMillis = System.currentTimeMillis();
-    
     //prevents adding empty comments to datastore
-    if (!comment.isEmpty()) {
-      Entity commentEntity = new Entity("Comment");
-      commentEntity.setProperty("value", comment);
-      commentEntity.setProperty("timeMillis", timeMillis);
-
+    if (!request.getParameter("Comment:").isEmpty()) {
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      datastore.put(commentEntity);
+      datastore.put(convertToEntity(request));
     }
     response.sendRedirect("/comments.html");
   }
 
-  public List<Comment> getDatastoreComments(int maxComments) {
-    List<Comment> comments = new ArrayList<>();
+  public Entity convertToEntity(HttpServletRequest request) {
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("value", request.getParameter("Comment:"));
+    commentEntity.setProperty("name", request.getParameter("Name:"));
+    commentEntity.setProperty("timeMillis", System.currentTimeMillis());
+
+    return commentEntity;
+  }
+
+  public Map<String, List<Comment>> getDatastoreComments(int maxComments) {
+    Map<String, List<Comment>> commentsByName = new HashMap<>();
+
     Query query = new Query("Comment").addSort("timeMillis", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    
+
     datastore.prepare(query).asList(FetchOptions.Builder.withLimit(maxComments)).forEach((entity)-> {
-      comments.add(Comment.entityToComment(entity));
+      String name = (String) entity.getProperty("name");
+
+      if(!commentsByName.containsKey(name)) {
+        commentsByName.put(name, new ArrayList<Comment>());
+      }
+      commentsByName.get(name).add(Comment.fromEntity(entity));
     });
-    return comments;
+
+    return commentsByName;
```

Comment:
I think from the user's perspective it make more sense to see comments ordered by time rather than grouped by user.

Up to you though!